parcels/field.py

```diff
@@ -945,6 +949,9 @@ def interpolator3D(self, ti, z, y, x, time):
                 xsi*(1-eta) * data[yi, xi+1] + \
                 xsi*eta * data[yi+1, xi+1] + \
                 (1-xsi)*eta * data[yi+1, xi]
+            if self.gridindexingtype == 'pop' and zi >= self.grid.zdim-2:
+                # Since POP is indexed at cell top, allow linear interpolation of W to zero in upper cell)
```

Comment:
Read the POP manual section 5.2 and 5.2.1: http://www.cesm.ucar.edu/models/cesm1.2/pop2/doc/sci/POPRefManual.pdf . I understand that W at the top layer changes the sea surface height anomaly \eta, and no tracers can be transported through the surface z=\eta. Is the best thing to interpolate between bottom W and W==0 at z==\eta (instead of z==0) in the upper cell?

Reply:
Sorry, this was a type-o in the comment. It has to be "Since POP is indexed at cell top, allow linear interpolation of W to zero in _lowest_ cell". That's also what the condition `zi >= self.grid.zdim-2` is for. I've now fixed the comment text